index.js

```diff
@@ -1 +1,123 @@
-console.log("hello")
\ No newline at end of file
+
+function createCountry(country) {
+
+    const countryLink = document.createElement("a");
+    countryLink.href = `details.html?country=${encodeURIComponent(country.name.common)}`;
+    countryLink.className = "country scale-effect";
+    countryLink.setAttribute("data-country-name", country.name.common);
+
+    const countryFlagDiv = document.createElement("div");
+    countryFlagDiv.className = "country-flag";
+    const flagImg = document.createElement("img");
+    flagImg.src = country.flags.png;
+    flagImg.alt = `${country.name} Flag`;
+    countryFlagDiv.appendChild(flagImg);
+
+    const countryInfoDiv = document.createElement("div");
+    countryInfoDiv.className = "country-info";
+
+    const titleElement = document.createElement("h2");
+    titleElement.className = "country-title";
+    titleElement.textContent = country.name.common;
+
+    const ulElement = document.createElement("ul");
+    ulElement.className = "country-brief";
+
+    const populationLi = document.createElement("li");
+    populationLi.innerHTML = `<strong>Population: </strong>${country.population}`;
+    const regionLi = document.createElement("li");
+    regionLi.innerHTML = `<strong>Region: </strong>${country.region}`;
+    const capitalLi = document.createElement("li");
+    capitalLi.innerHTML = `<strong>Capital: </strong>${country.capital}`;
+
+    ulElement.appendChild(populationLi);
+    ulElement.appendChild(regionLi);
+    ulElement.appendChild(capitalLi);
+
+    countryInfoDiv.appendChild(titleElement);
+    countryInfoDiv.appendChild(ulElement);
+
+    countryLink.appendChild(countryFlagDiv);
+    countryLink.appendChild(countryInfoDiv);
+
+    countryLink.addEventListener("click", () => {
+        window.location.href = countryLink.href;
+    });
+
+    return countryLink;
+}
+
+let countries = [];
+
+const requestCountries = async () => {
+    try {
+        const response = await fetch("https://restcountries.com/v3.1/all");
+        if (!response.ok) {
+            throw new Error('Network response was not ok.');
+        }
+        const data = await response.json();
+        countries = data;
+    
+        const countriesGrid = document.querySelector(".countries-grid");
+    
+        countries.forEach(country => {
+            const countryBlock = createCountry(country);
+            countriesGrid.appendChild(countryBlock);
+        });
+    } catch (error) {
+        console.error('Error fetching country data:', error);
+    }
+ }
+
+requestCountries();
+
+const dropdownWrapper = document.querySelector('.dropdown-wrapper');
```

Comment:
Consider wrapping this code to a function and call it right away, this will make the code more understandable (instead of just manipulating some elements